chapter2/tests/t19/terraform/base/main.tf

```diff
@@ -0,0 +1,23 @@
+module "nginx" {
+  source                       = "../_modules/nginx"
+  container_image              = var.nginx.image
+  container_name               = var.nginx.container_name != "" ? var.nginx.container_name : null
+  container_ports              = var.nginx.container_ports != "" ? var.nginx.container_ports : null
+  container_volumes            = var.nginx.container_volumes
+  container_image_keep_locally = var.nginx.keep_locally
+  client                       = var.client
+  project                      = var.project
+  environment                  = var.environment
+}
+
+module "redis" {
+  source                       = "../_modules/container"
+  count                        = var.use_redis ? 1 : 0
+  container_image              = var.redis.image
+  container_name               = var.redis.container_name
+  container_ports              = var.use_redis ? var.redis.container_ports : null
```

Comment:
make all validation in the inner module, not in the outer invocation. That way if your module REQUIRE the port - you will get error

Reply:
> make all validation in the inner module, not in the outer invocation. That way if your module REQUIRE the port - you will get error

@dmitry-mightydevops
Done